src/crypto/tls/cipher_suites.go

```diff
@@ -156,6 +164,31 @@ func (f *fixedNonceAEAD) Open(out, nonce, plaintext, additionalData []byte) ([]b
 	return f.aead.Open(out, f.openNonce, plaintext, additionalData)
 }
 
+func (f *xoredNonceAEAD) NonceSize() int { return 12 }
```

Comment:
Yes in that case it is 8.  8 are passed to xoredNonceAEAD that are then made into 12 for ChaCha-Poly.